docs/.vuepress/plugins/chart-editor/components/ChartEditor.vue

```diff
@@ -13,15 +13,18 @@
 </template>
 
 <script>
-import {Chart} from 'chart.js';
+import {Chart, registerables} from 'chart.js';
+import {merge} from 'chart.js/helpers';
 import * as Utils from '../utils';
 
 // Components
 import ChartActions from './ChartActions.vue';
 import ChartView from './ChartView.vue';
 import CodeEditor from './CodeEditor.vue';
 
-const CHART_DEFAULTS = Chart.helpers.merge({}, Chart.defaults);
+Chart.register(...registerables);
+
+const CHART_DEFAULTS = merge({}, Chart.defaults);
```

Comment:
@simonbrunel Having a breakpoint there, it seems the plugin defaults are present even if I don't move the plugin registration. Moving it to the ChartEditor component doesn't fix the problem. Looks to me you are on the right path though, it seems that the config is not containing the default values when rendering the labels.

Reply:
The plugin is not registered at that point?

Reply:
I can see it in the registry, looks to me like it is:
![Screenshot 2021-02-24 at 09 19 55](https://user-images.githubusercontent.com/2706307/108968769-8211de80-7681-11eb-8f5c-bce7925de2f3.png)

Reply:
It should, from the `ChartView.vue` import.

Reply:
Looks like the plugin is registered in the view. So I don't think the `CHART_DEFAULTS` contain `plugins.datalabels` (what I should have asked).

Reply:
There are few things changed in the proxied options in core since beta.11, would be interesting to see if those make a difference. I think we can push next release quite soon.

Reply:
Seems to be fixed if I link the latest Chart.js master. Eagerly awaiting for next beta, as usual 😄

Reply:
Updated to latest beta and removed the descriptors. Tests are passing, but the default values still seem not to be picked up in the docs. Will look into it a bit more, but I'd appreciate any help, not a pro with Vue.js...

Reply:
I suspect that proxied options are not enumerable, thus not mergeable which would fail [resolving label options](https://github.com/chartjs/chartjs-plugin-datalabels/blob/50bb58e8c90fc32d048009763997a35a7f15d634/src/plugin.js#L28). It's probably not related to the docs but I think unit tests always explicitly set options to test in the chart config and so never fallback to defaults. And for the [`defaults` unit test](https://github.com/chartjs/chartjs-plugin-datalabels/blob/50bb58e8c90fc32d048009763997a35a7f15d634/types/test/options.default.ts), we test before resolving label options, so it works.

Reply:
Probably not it, but this part needs to be updated: 
https://github.com/chartjs/chartjs-plugin-datalabels/blob/9586e39f56b9a5c6dc733d9bb4888b6147e90140/docs/.vuepress/config.js#L20-L32

Reply:
@simonbrunel is half right. The proxied options are enumerable, but they do not contain the defaults. So the fix is probably to add the defaults in [that merge](https://github.com/chartjs/chartjs-plugin-datalabels/blob/50bb58e8c90fc32d048009763997a35a7f15d634/src/plugin.js#L28)

Its not compatible with Chart.js v2, but it gives the plugin maintainers even more control on how the options should be handled.

Reply:
It's possible to access the default options from this passed `options` proxied object, but these default values are not enumerable. I don't think it's the correct behavior. For example, accessing `options.display` will return `true/false` from the defaults, but `{...options}` will not contain `display`, which I would expect to be there.

Reply:
> So the fix is probably to add the defaults in that merge

I don't think that the correct fix. Instead, we should make sure that these default options are enumerable as part of the proxied `options`. It's currently inconsistent between accessing the options property directly and iterating other the object keys. Plugin developers should not have to care about the internal implementation of this `options` object: it should behave as a plain object as if config were merged with defaults.

Reply:
Oh, right. Currently `Object.keys` return the keys in the topmost options object, while `Object.getOwnPropertyNames` returns all the inherited keys. The reasoning behind that might not be valid anymore, have to take a look.

Reply:
https://github.com/chartjs/Chart.js/pull/8553 should fix this

Reply:
Tested it with Chart.js master, seems to be working now! Thanks @kurkle & @simonbrunel!